File: packages/rooms/create/create.go
package main

import (
	"context"
	"github.com/go-redis/redis/v8"
	"os"
	"time"
)

type player struct{
	id string
	authorization_key string
	username string
	owner bool
}

func Main(args map[string]interface{}) map[string]interface{} {
	redis_option, _ := redis.ParseURL(os.Getenv("DATABASE_URL"))
	rdb := redis.NewClient(redis_option)
	room_template := make(map[string]interface{})
	room_template["id"] = args["id"].(string)
	room_template["players"] = make([]player, 0)
	room_template["creation_time"] = time.Now().Unix()
	room_template["bomb_start_time"] = 0
	room_template["words"] = make([]string, 0)
	room_template["language"] = args["language"].(string)
	room_template["current_word"] = ""
	room_template["started"] = false
	room_template["finished"] = false

	rdb.Set(context.Background(), "room:"+args["room_id"].(string), room_template, 0)
	msg := make(map[string]interface{})
	msg["body"] = "Room created"
	return msg
}

File: packages/rooms/create/create.js
const createClient = require('redis').createClient;
require('dotenv').config()

async function main(args) {
    const redis = createClient({
        url: process.env.DATABASE_URL
    });
    await redis.connect();
    room_template = {}
    room_template.id = args.id
    room_template.players = []
    room_template.creation_time = Date.now()
    room_template.bomb_start_time = 0
    room_template.words = []
    room_template.language = args.language
    room_template.current_word = ""
    room_template.password = args.password
    room_template.started = false
    room_template.finished = false
    await redis.set(args.id, JSON.stringify(room_template) )
    redis.disconnect();
    return {}
}
main()
exports.main = main;